### PR TITLE
Support asynchronous logging.

### PR DESCRIPTION
**Pull request by frankfliu**
Before or while filing an issue please feel free to join our [ slack channel](https://join.slack.com/t/mms-awslabs/shared_invite/enQtNDk4MTgzNDc5NzE4LTBkYTAwMjBjMTVmZTdkODRmYTZkNjdjZGYxZDI0ODhiZDdlM2Y0ZGJiZTczMGY3Njc4MmM3OTQ0OWI2ZDMyNGQ) to get in touch with development team, ask questions, find out what's cooking and more!

## Testing done:

**To run CI tests on your changes refer [README.md](https://github.com/awslabs/mxnet-model-server/blob/master/ci/README.md)**

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -153,6 +153,7 @@ Most of those properties are designed for performance tuning. Adjusting those nu
 * netty_client_threads: number of backend netty thread, default: number of logical processors available to the JVM.
 * default_workers_per_model: number of workers to create for each model that loaded at startup time, default: available GPUs in system or number of logical processors available to the JVM.
 * job_queue_size: number inference jobs that frontend will queue before backend can serve, default 100.
+* async_logging: enable asynchronous logging, default false.
```
Comment (ddavydenko) [inline docs/configuration.md]: I would suggest to add some more description here regarding what is the logging behavior by default (that it is rotational) and why/when async could be useful instead.

Also, I think ./docs/logging.md needs to be updated to specify that multiple logging modes are supported with details on what each of them (rotational -vs- async) does.

File: frontend/server/src/main/java/com/amazonaws/ml/mms/util/ConfigManager.java [hunk 2]
```diff
@@ -445,6 +453,33 @@ private File findMmsHome() {
...
+        enableAsyncLogging(Logger.getLogger(MODEL_LOGGER));
+        enableAsyncLogging(Logger.getLogger(MMS_METRICS_LOGGER));
+        enableAsyncLogging(Logger.getLogger("ACCESS_LOG"));
+        enableAsyncLogging(Logger.getLogger("com.amazonaws.ml.mms"));
+    }
+
+    private void enableAsyncLogging(Logger logger) {
+        AsyncAppender asyncAppender = new AsyncAppender();
+
+        @SuppressWarnings("unchecked")
```
Comment (vrakesh) [inline frontend/server/src/main/java/com/amazonaws/ml/mms/util/ConfigManager.java]: Why is the warning suppressed here ?

Comment (frankfliu) [reply frontend/server/src/main/java/com/amazonaws/ml/mms/util/ConfigManager.java]: Warning is caused by log4j API, logger.getAllAppenders() didn't return parameterized Enumeration. We have to suppress it here.

**Comment by vdantu** [comment]
@vrakesh and @ddavydenko : Anything else needs to change for this PRs approval? This is a good change to have in mainline master and our "pre released" pip packages.

**Pull request merged**